scripts/drift-makeproducts

```diff
@@ -46,52 +46,138 @@ def queue_config(args):
+
+    # Set up required PBS vars
+    if 'nodes' not in conf:
+        raise Exception('Nodes is required.')
+    clusterconf['nodes'] = conf['nodes']
+
+    if 'time' not in conf:
+        raise Exception('Job time is required.')
+    clusterconf['time'] = conf['time']
+
+    # Set queueing system w. defaults
+    cluster_defaults = {
+        'pbs' : {'ppn': 8, 'mem': '16000M', 'account':None, 'submit':'qsub',},
+        'slurm': {'ppn': 32, 'mem': '0', 'account':None, 'submit':'sbatch',}
+    }
+
+    if queue_sys in cluster_defaults:
+        
+        clusterconf['queue_sys'] = queue_sys
+        clusterconf['ppn'] = conf['ppn'] if 'ppn' in conf else cluster_defaults[queue_sys]['ppn']
+        clusterconf['mem'] = conf['mem'] if 'mem' in conf else cluster_defaults[queue_sys]['mem']
+        clusterconf['account'] = conf['account'] if 'account' in conf else cluster_defaults[queue_sys]['account']
+
+        submit_command = conf['submit_command'] if 'submit_command' in conf else cluster_defaults[queue_sys]['submit']
+        
+    else:
+        clusterconf['queue_sys'] = queue_sys
+        clusterconf['ppn'] = conf['ppn'] if 'ppn' in conf else 8
+        clusterconf['mem'] = conf['mem'] if 'mem' in conf else '0'
+        clusterconf['account'] = conf['account'] if 'account' in conf else None
+
+        if 'submit_command' in conf:
+            submit_command = conf['submit_command']
+        else:
+            raise Exception('Need to specify submit command for unknown scheduler.')
+        
+        if 'script_template' in conf:
+            script_templates[queue_sys] = conf['script_template']
+        else:
+            raise Exception('Need to specify submit script string for unknown scheduler.')
+        
+    # Set up optional PBS vars
+    clusterconf['ompnum'] = conf['ompnum'] if 'ompnum' in conf else 8
+    clusterconf['queue'] = conf['queue'] if 'queue' in conf else 'batch'
+    clusterconf['pernode'] = conf['pernode'] if 'pernode' in conf else 1
+    clusterconf['name'] = conf['name'] if 'name' in conf else 'job'
+
+    # Set vars only needed to create script
+    clusterconf['mpiproc'] = clusterconf['nodes'] * clusterconf['pernode']
+    clusterconf['workdir'] = outdir
+    clusterconf['scriptpath'] = os.path.realpath(__file__)
+    clusterconf['logpath'] = submitdir + '/jobout.log'
+    clusterconf['configpath'] = submitdir + '/config.yaml'
+
+    # Set up virtualenv
+    if 'venv' in conf:
+        if not os.path.exists(conf['venv'] + '/bin/activate'):
+            raise Exception('Could not find virtualenv')
+
+        clusterconf['venv'] = conf['venv'] + '/bin/activate'
+    else:
+        clusterconf['venv'] = '/dev/null'
+
+    script = script_templates[queue_sys] % clusterconf
+
+    scriptname = submitdir + "/jobscript.sh"
 
-    script="""#!/bin/bash
+    with open(scriptname, 'w') as f:
+        f.write(script)
+
+    if not args.nosubmit:
+        os.system('cd %s; ' % submitdir + submit_command + ' jobscript.sh')
```

Comment:
Could you merge this into a single formatted string? The operator precedence isn't immediately memorable to me, and it took me a while to realise that is was correct.